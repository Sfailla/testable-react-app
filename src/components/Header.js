import React from 'react';
import logo from '../svg/react-logo.svg';

const Header = () => {
	return <div />;
};

export default Header;
