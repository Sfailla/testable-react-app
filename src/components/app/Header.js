import React from 'react';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles(theme => ({
	header: {
		width: '100%',
		height: '50px',
		backgroundColor: theme.palette.tertiary.main,
		padding: theme.spacing(0, 1),
		color: theme.palette.common.white,
		boxShadow: theme.shadows[5],

		position: 'absolute',
		top: '0',
		left: '0'
	},
	navigation: {
		display: 'flex',
		fontWeight: 'bold',
		'& p:not(:last-of-type)': {
			paddingRight: theme.spacing(1)
		}
	},
	wrapper: {
		width: '100%',
		height: '50px',
		display: 'flex',
		justifyContent: 'space-between',
		alignItems: 'center'
	}
}));

const Header = () => {
	const classes = useStyles();
	return (
		<header className={classes.header}>
			<div className={classes.wrapper}>
				<h2>Logo</h2>
				<div className={classes.navigation}>
					<p>Home</p>
					<p>About</p>
				</div>
			</div>
		</header>
	);
};

export default Header;
