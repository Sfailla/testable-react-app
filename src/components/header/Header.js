import React from 'react';
import Navigation from '../nav/Navigation';
import { useStyles } from './HeaderStyles';

function Header() {
	const classes = useStyles();
	return (
		<header className={classes.header}>
			<div className={classes.wrapper}>
				<h2>Logo</h2>
				<div className={classes.navigation}>
					<Navigation />
				</div>
			</div>
		</header>
	);
}

export default Header;
