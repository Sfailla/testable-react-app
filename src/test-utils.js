import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'react-jss';
import { theme } from './styles/theme';

const AllMyProviders = ({ children }) => {
	return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const MyCustomRender = (component, options) =>
	render(component, { wrapper: AllMyProviders, ...options });

// re-export everything from the `react-testing-library`
export * from '@testing-library/react';

// export our custom render method
export { MyCustomRender as render };
