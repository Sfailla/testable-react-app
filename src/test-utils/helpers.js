import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'react-jss';
import { theme } from '../styles/theme';
import { StaticRouter } from 'react-router-dom';

function shallowWithTheme(child) {
	return shallow(child, {
		wrappingComponent: ({ children }) => (
			<ThemeProvider theme={theme}>{children}</ThemeProvider>
		)
	});
}

function mountWithTheme(child) {
	return mount(child, {
		wrappingComponent: ({ children }) => (
			<ThemeProvider theme={theme}>{children}</ThemeProvider>
		)
	});
}

function wrapWithRouter({ children }) {
	return <StaticRouter>{children}</StaticRouter>;
}

module.exports = {
	shallowWithTheme,
	mountWithTheme,
	wrapWithRouter
};
