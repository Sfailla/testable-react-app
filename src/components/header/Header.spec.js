import React from 'react';
import { render, renderWithRouter } from '../../test-utils';

import Header from './Header';

describe('<Header /> unit tests', () => {
	it('should render <Header /> correctly', () => {
		renderWithRouter(<Header />);
	});

	it('should contain h2 tag with correct text value', () => {
		const { getByText } = render(renderWithRouter(<Header />));
		expect(getByText(/logo/i)).toBeInTheDocument();
	});

	it('should contain <header> element', () => {
		const { getByTestId } = render(renderWithRouter(<Header />));
		const header = getByTestId('header');
		expect(header).toBeInTheDocument();
	});

	it('should take a snapshot of component', () => {
		const { container } = render(renderWithRouter(<Header />));
		expect(container).toMatchSnapshot();
	});
});
