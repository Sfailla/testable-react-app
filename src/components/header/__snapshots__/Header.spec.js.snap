// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> unit tests should take a snapshot of component 1`] = `
<div>
  <header
    class="header-0-2-1"
    data-testid="header"
  >
    <div
      class="wrapper-0-2-3"
    >
      <h2>
        Logo
      </h2>
      <div
        class="navigation-0-2-2"
      >
        <nav>
          <ul
            class="list-0-2-4"
          >
            <li
              class="listItem-0-2-5"
            >
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="listItem-0-2-5"
            >
              <a
                href="/forms"
              >
                Forms
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;
